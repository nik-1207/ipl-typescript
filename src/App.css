.loader{
    margin: 25% 45%;

}