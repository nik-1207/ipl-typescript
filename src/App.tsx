import { Route, BrowserRouter as Router, Switch } from "react-router-dom";

import "./App.css";
import Navbar from "./components/NavBar";
import AllTeams from "./pages/AllTeams";
import TeamDetails from "./pages/TeamDetails";

function App() {
  return (
    <div className="App">
      <Navbar />
      <Router>
        <Switch>
          <Route path="/" exact component={AllTeams} />
          <Route path="/teams/:url" exact component={TeamDetails} />
        </Switch>
      </Router>
    </div>
  );
}

export default App;
