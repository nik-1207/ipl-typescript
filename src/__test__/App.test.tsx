import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import App from "../App";
jest.mock("../pages/AllTeams", () => () => (
  <div>Hello World from AllTeams</div>
));

it("renders correctly", async () => {
  render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
  const data = await screen.getByText("Hello World from AllTeams");
  expect(data).toBeInTheDocument();
});
