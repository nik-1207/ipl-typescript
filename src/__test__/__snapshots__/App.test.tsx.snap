// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <nav
    className="navbar-0-2-2"
    title="navbar"
  >
    <img
      alt="ipl-logo"
      className="logoImage-0-2-1"
      src="ipl-logo.png"
    />
    <h1
      className="navHeading-0-2-3"
      title="heading"
    >
      INDIAN PREMIRE LEAGUE
    </h1>
  </nav>
  <div
    className="loader"
  >
    <div
      aria-busy="true"
      className=""
      style={Object {}}
    >
      <svg
        aria-label="audio-loading"
        height={80}
        viewBox="0 0 38 38"
        width={80}
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <linearGradient
            id="a"
            x1="8.042%"
            x2="65.682%"
            y1="0%"
            y2="23.865%"
          >
            <stop
              offset="0%"
              stopColor="#00BFFF"
              stopOpacity="0"
            />
            <stop
              offset="63.146%"
              stopColor="#00BFFF"
              stopOpacity=".631"
            />
            <stop
              offset="100%"
              stopColor="#00BFFF"
            />
          </linearGradient>
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            transform="translate(1 1)"
          >
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
              id="Oval-2"
              stroke="#00BFFF"
              strokeWidth="2"
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
            <circle
              cx="36"
              cy="18"
              fill="#fff"
              r={1}
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </circle>
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
